### PR TITLE
Display more calculations

### LINKED ISSUE #935: Display more calculations on screen
#### Problem you'd like to fix
Currently, there can be only 3 calculations display on screen.

#### Screenshots
Here is what currently happens: 
SingleLine0

This is what the Ti-Nspire CX and HP Prime do:
SingleLine SingleLine2

#### Describe the solution you'd like
Calculations + answers that can fit on a single line should be displayed on a single line while longer calculations + answers should be displayed on two lines (like how all calculations + answers are currently displayed). I don't think this could be too complicated to implement.

#### Describe alternatives you've considered
There could also be a font size option (small - medium - large) #937?

### PR DESCRIPTION
**Pull request by 0b101**
This adds support for single lined calculations and fixed #935.
moCalc mmmmm
This was developed by @PiGuy2 and @0b101 

Here is a ROM dump: [epsilon.bin](https://tiplanet.org/modules/archives/download.php?id=2244957)
Flash using: [webdfu](https://ti-planet.github.io/webdfu_numworks/)

### CODE REVIEW

**Comment by PiGuy2** [comment]
As I mentioned in #935, there is still a small problem with the rendering, so this is not quite ready to merge.

File: apps/calculation/history_controller.cpp [hunk 1]
```diff
@@ -150,7 +150,7 @@ KDCoordinate HistoryController::rowHeight(int j) {
   }
   Calculation * calculation = m_calculationStore->calculationAtIndex(j);
   App * calculationApp = (App *)app();
-  return calculation->height(calculationApp->localContext()) + 4 * Metric::CommonSmallMargin;
+  return calculation->height(calculationApp->localContext()); //+ 2 * Metric::CommonSmallMargin;
```
Comment (0b101) [inline apps/calculation/history_controller.cpp]: Remove comment
```suggestion
  return calculation->height(calculationApp->localContext());
```

File: apps/calculation/calculation.cpp [hunk 2]
```diff
@@ -60,19 +60,44 @@ KDCoordinate Calculation::height(Context * context) {
...
+      }
     } else {
       KDCoordinate approximateOutputHeight = approximateLayout.layoutSize().height();
       KDCoordinate exactOutputHeight = exactLayout.layoutSize().height();
       KDCoordinate outputHeight = max(exactLayout.baseline(), approximateLayout.baseline()) + max(exactOutputHeight-exactLayout.baseline(), approximateOutputHeight-approximateLayout.baseline());
-      m_height = inputHeight + outputHeight;
+      KDCoordinate exactOutputWidth = exactLayout.layoutSize().width();
+      KDCoordinate approximateOutputWidth = approximateLayout.layoutSize().width();
+      KDCoordinate outputWidth = exactOutputWidth + approximateOutputWidth;
+      singleLine = outputWidth + inputWidth < maxWidth - 70;
```
Comment (0b101) [inline apps/calculation/calculation.cpp]: This should fix the bug with 11^9
```suggestion
      singleLine = outputWidth + inputWidth < maxWidth - 40;
```

File: apps/calculation/history_controller.cpp (hunk 1 above)
Comment (PiGuy2) [reply apps/calculation/history_controller.cpp]: This is fixed by 6b93fad

**Comment by 0b101** [comment]
> As I mentioned in #935, there is still a small problem with the rendering, so this is not quite ready to merge.

@PiGuy2 The rendering issues appear to be resolved in https://github.com/PiGuy2/epsilon/pull/6

**Comment by adriweb** [comment]
Nice ! (Also it made [the news](https://tiplanet.org/forum/viewtopic.php?f=97&t=22625) on TI-Planet :P)
Can you rebase (and fix conflicts)? Probably some commits could be squashed together too.

**Comment by 0b101** [comment]
This commit will be compatible with master/11.0.0 (without conflicts) once the fork is synced/updated and https://github.com/PiGuy2/epsilon/pull/7 is applied.

**Comment by PiGuy2** [comment]
This is now compatible with master, and tested pretty well!

**Comment by 0b101** [comment]
I posted a new ROM dump onto TI-Planet if anyone want's to test this out. 
ROM dump: [epsilon.bin](https://tiplanet.org/modules/archives/download.php?id=2244957)
Flash using: [webdfu](https://ti-planet.github.io/webdfu_numworks/)

**Comment by 0b101** [comment]
There is currently a display bug with the expanded view. It can be fixed by adding the following to lines 108-110: 
`if (expanded) {`
`      *memoizedHeight = *memoizedHeight + singleMargin;`
`}`

File: apps/calculation/calculation.cpp [hunk 3]
```diff
@@ -66,20 +66,44 @@ KDCoordinate Calculation::height(Context * context, bool expanded) {
...
+      if (singleLine) {
+        *memoizedHeight  = (inputHeight >= outputHeight) ? inputHeight + singleMargin : outputHeight + singleMargin;
+      } else {
+        *memoizedHeight  = inputHeight + outputHeight + doubleMargin;
+      }
     }
     /* For all display output except ExactAndApproximateToggle, the selected
```
Comment (0b101) [inline apps/calculation/calculation.cpp]:
```suggestion
     if (expanded) {
         *memoizedHeight = *memoizedHeight + singleMargin;
     }
     /* For all display output except ExactAndApproximateToggle, the selected
```

Comment (PiGuy2) [reply apps/calculation/calculation.cpp]: I compiled it with this change, but I'm not sure what bug it is fixing.

Comment (0b101) [reply apps/calculation/calculation.cpp]: There is a bug with the expanded view where it either doesn’t display at all or it displays incorrectly. Try entering something like `0.8/0.3` and then pressing the up arrow after executing. After the change, it should correctly display the fractional answer along side the decimal answer when selected.

**Comment by 0b101** [comment]
This is just a quick update: I am currently rewriting this to be less buggy and optional (with a display style setting of normal or condensed).

**Comment by 0b101** [comment]
I’m closing this so I can upload my new version (from Omega) tomorrow

**Pull request closed without merging**